executor/admin.go

```diff
@@ -295,15 +294,11 @@ func (e *RecoverIndexExec) buildTableScan(ctx context.Context, txn kv.Transactio
 func buildRecoverIndexKeyRanges(sctx *stmtctx.StatementContext, tid int64, startHandle kv.Handle) ([]kv.KeyRange, error) {
 	var startKey []byte
 	if startHandle == nil {
-		startKey = tablecodec.EncodeRowKey(tid, []byte{codec.NilFlag})
+		startKey = tablecodec.GenTableRecordPrefix(tid).Next()
 	} else {
-		startKey = tablecodec.EncodeRowKey(tid, startHandle.Next().Encoded())
+		startKey = tablecodec.EncodeRowKey(tid, startHandle.Encoded()).PrefixNext()
 	}
-	maxVal, err := codec.EncodeKey(sctx, nil, types.MaxValueDatum())
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	endKey := tablecodec.EncodeRowKey(tid, maxVal)
+	endKey := tablecodec.GenTableRecordPrefix(tid).PrefixNext()
```

Comment:
suppose there exists lots of index keys between `(t_1, t_2)`. will this range scan index keys?

Reply:
No, it does't scan index keys because `GenTableRecordPrefix` represent `"t[tableID]_r"`. So the `PrefixNext` is `"t[tableID]_s"`.